File: service/platform-service/example/task/flink_task.py
from pyflink.datastream import StreamExecutionEnvironment
from pyflink.datastream.connectors import FlinkKafkaConsumer, FlinkKafkaProducer
from pyflink.common.serialization import SimpleStringSchema
from pyflink.datastream.functions import MapFunction
import json


class MessageProcessor(MapFunction):
    def map(self, value):
        try:
            data = json.loads(value)
            message = data.get("message", "")
            result = f"You sent {message}"
            return json.dumps({"response": result})
        except json.JSONDecodeError:
            return json.dumps({"error": "Invalid JSON input"})


def main():
    env = StreamExecutionEnvironment.get_execution_environment()
    env.set_parallelism(1)
    
    # Example Kafka source configuration (adjust as needed)
    kafka_props = {
        'bootstrap.servers': 'localhost:9092',
        'group.id': 'flink-python-consumer'
    }
    
    # Create Kafka consumer
    kafka_consumer = FlinkKafkaConsumer(
        topics='input-topic',
        deserialization_schema=SimpleStringSchema(),
        properties=kafka_props
    )
    
    # Create Kafka producer
    kafka_producer = FlinkKafkaProducer(
        topic='output-topic',
        serialization_schema=SimpleStringSchema(),
        producer_config=kafka_props
    )
    
    # Process stream
    data_stream = env.add_source(kafka_consumer)
    
    processed_stream = data_stream.map(MessageProcessor())
    
    processed_stream.add_sink(kafka_producer)
    
    # Execute the job
    env.execute("Python Flink Message Processor")


if __name__ == "__main__":
    main()

File: service/platform-service/example/task/FlinkTask.java
import org.apache.flink.api.common.functions.MapFunction;
import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer;
import org.apache.flink.streaming.connectors.kafka.FlinkKafkaProducer;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;
import java.util.Properties;

public class FlinkTask {
    
    public static class MessageProcessor implements MapFunction<String, String> {
        private final ObjectMapper objectMapper = new ObjectMapper();
        
        @Override
        public String map(String value) throws Exception {
            try {
                ObjectNode inputJson = (ObjectNode) objectMapper.readTree(value);
                String message = inputJson.get("message").asText("");
                
                ObjectNode outputJson = objectMapper.createObjectNode();
                outputJson.put("response", "You sent " + message);
                
                return objectMapper.writeValueAsString(outputJson);
            } catch (Exception e) {
                ObjectNode errorJson = objectMapper.createObjectNode();
                errorJson.put("error", "Invalid JSON input");
                return objectMapper.writeValueAsString(errorJson);
            }
        }
    }
    
    public static void main(String[] args) throws Exception {
        // Set up the execution environment
        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);
        
        // Kafka properties
        Properties kafkaProps = new Properties();
        kafkaProps.setProperty("bootstrap.servers", "localhost:9092");
        kafkaProps.setProperty("group.id", "flink-java-consumer");
        
        // Create Kafka consumer
        FlinkKafkaConsumer<String> kafkaConsumer = new FlinkKafkaConsumer<>(
            "input-topic",
            new SimpleStringSchema(),
            kafkaProps
        );
        
        // Create Kafka producer
        FlinkKafkaProducer<String> kafkaProducer = new FlinkKafkaProducer<>(
            "output-topic",
            new SimpleStringSchema(),
            kafkaProps
        );
        
        // Create data stream from Kafka
        DataStream<String> inputStream = env.addSource(kafkaConsumer);
        
        // Process the stream
        DataStream<String> processedStream = inputStream.map(new MessageProcessor());
        
        // Send processed data to Kafka
        processedStream.addSink(kafkaProducer);
        
        // Execute the Flink job
        env.execute("Java Flink Message Processor");
    }
}